src/ess/sans/masking.py

```diff
@@ -12,33 +12,15 @@
 from .types import (
     MaskedData,
     MaskedDetectorIDs,
-    PixelMask,
     PixelMaskFilename,
-    RawData,
-    SampleRun,
     ScatteringRunType,
+    TofData,
 )
 
 
-def to_pixel_mask(data: RawData[SampleRun], masked: MaskedDetectorIDs) -> PixelMask:
-    """Convert a list of masked detector IDs to a pixel mask.
-
-    Parameters
-    ----------
-    data:
-        Raw data, defining the detector IDs.
-    masked:
-        The masked detector IDs.
-    """
-    ids = data.coords['detector_number']
-    mask = sc.zeros(sizes=ids.sizes, dtype='bool')
-    mask.values[np.isin(ids.values, masked.values)] = True
-    return PixelMask(mask)
-
-
 def apply_pixel_masks(
-    data: RawData[ScatteringRunType],
-    masks: Optional[sciline.Series[PixelMaskFilename, PixelMask]],
+    data: TofData[ScatteringRunType],
+    masked_ids: Optional[sciline.Series[PixelMaskFilename, MaskedDetectorIDs]],
 ) -> MaskedData[ScatteringRunType]:
```

Comment:
Not totally happy with this change, as it will prevent using manual masks that are constructed, e.g., using a condition of the pixel positions, etc. Can the `MaskedDetectorIDs` -> `PixelMask` conversion be done in a separate step?

Reply:
The issue was that it was using the `RawData[SampleRun]` to get the detector ids, and using it also for the `BackgroundRun`. This led to an infinite recursion error when trying to set a parameter table for using events from multiple runs, I think because it was trying to get a `sciline.Series[Filename[BackgroundRun], RawData[SampleRun]]` which it could not materialize.

I could have a `PixelMask[ScatteringRunType]` as an alternative? I think that would work?

Reply:
> it will prevent using manual masks that are constructed, e.g., using a condition of the pixel positions

I think in this case you would have to use a different provider anyway, because the masks are `sciline.Series[PixelMaskFilename, MaskedDetectorIDs]`, i.e. they depend on `PixelMaskFilename`, so most probably wouldn't be created from a condition?

FYI I'm still running into issues with `PixelMask[ScatteringRunType]`.